File: tests/integration/test_enhanced_parser.js
import EnhancedWorkflowParser from './parsers/workflow/workflowParser_enhanced.js';

const parser = new EnhancedWorkflowParser();

const testPrompts = [
    "Generate KYC proof for Bob",
    "Generate KYC proof for Bob and verify on Solana",
    "Generate KYC proof for Bob and verify on Solana, then send 0.05 USDC on Solana",
    "Generate KYC proof for Bob, verify it on Solana, then send him 0.05 USDC",
    "Generate KYC proof for Bob and verify on Solana, then send him 0.05 USDC on Solana",
    "Send 0.05 USDC on Solana if Bob is KYC verified on Solana"
];

testPrompts.forEach(prompt => {
    console.log(`\n${'='.repeat(70)}`);
    console.log(`Prompt: ${prompt}`);
    console.log(`${'='.repeat(70)}`);
    
    try {
        const result = parser.parseCommand(prompt);
        console.log("\nParsed workflow:");
        console.log(`  Description: ${result.description}`);
        console.log(`  Requires Proofs: ${result.requiresProofs}`);
        console.log(`  Steps: ${result.steps.length}`);
        
        result.steps.forEach((step, i) => {
            console.log(`\n  Step ${i + 1}: ${step.type}`);
            console.log(`    Description: ${step.description}`);
            if (step.person) console.log(`    Person: ${step.person}`);
            if (step.proofType) console.log(`    Proof Type: ${step.proofType}`);
            if (step.recipient) console.log(`    Recipient: ${step.recipient}`);
            if (step.amount) console.log(`    Amount: ${step.amount}`);
            if (step.blockchain) console.log(`    Blockchain: ${step.blockchain}`);
            if (step.verificationType) console.log(`    Verification Type: ${step.verificationType}`);
        });
    } catch (error) {
        console.log("Error:", error.message);
        console.log(error.stack);
    }
});

File: tests/unit/test_enhanced_parser.py
#!/usr/bin/env python3
"""Test the enhanced OpenAI workflow parser"""

import os
import json
from parsers.workflow.openaiWorkflowParserEnhanced import EnhancedOpenAIWorkflowParser

# Test commands
test_commands = [
    "Generate KYC proof, verify it on blockchain, then send 0.1 USDC to Alice on Ethereum",
    "Generate location proof and verify on Solana, then transfer 0.05 USDC to Bob",
    "Generate AI content proof, verify locally and on Ethereum, then send 0.2 USDC to Charlie on Solana",
    "If Alice is KYC verified on-chain, send her 0.05 USDC on Solana",
]

# Get API key
api_key = os.getenv('OPENAI_API_KEY')
if not api_key:
    print("Error: OPENAI_API_KEY not set")
    exit(1)

# Create parser
parser = EnhancedOpenAIWorkflowParser(api_key)

# Test each command
for cmd in test_commands:
    print(f"\n{'='*60}")
    print(f"Command: {cmd}")
    print('='*60)
    
    try:
        result = parser.parse_workflow(cmd)
        
        # Print parsed steps
        print(f"\nParsed into {len(result.get('steps', []))} steps:")
        for i, step in enumerate(result.get('steps', [])):
            print(f"{i+1}. {step['type']}: {step.get('description', 'No description')}")
            if step['type'] in ['verify_on_ethereum', 'verify_on_solana']:
                print(f"   - Blockchain: {step['type'].split('_')[-1]}")
                print(f"   - Proof type: {step.get('proof_type', 'unknown')}")
        
        # Validate workflow
        is_valid = parser.validate_workflow(result)
        print(f"\nWorkflow valid: {is_valid}")
        
    except Exception as e:
        print(f"Error: {e}")